### PR TITLE
[FLINK-2125][streaming] Delimiter change from char to string

### PR DESCRIPTION
**Pull request by ajaybhat**

### CODE REVIEW

**Comment by rmetzger** [comment]
Looks like the tests are failing.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SocketTextStreamFunction.java [hunk 1]
```diff
@@ -84,25 +84,25 @@ public SocketTextStreamFunction(String hostname, int port, char delimiter, long
...
-					}
-					else {
-						// truncate trailing carriage return
-						if (delimiter == '\n' && buffer.length() > 0 && buffer.charAt(buffer.length() - 1) == '\r') {
+				char[] charBuffer = new char[Math.max(8192,delimiter.length()*2)];
+				int bytesRead;
+				while (isRunning && (bytesRead = reader.read(charBuffer)) != -1) {
+					String input = new String(charBuffer, 0, bytesRead);
+					int start = 0,pos;
+					while ((pos = input.indexOf(delimiter, start)) > -1) {
```
Comment (tillrohrmann) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SocketTextStreamFunction.java]: What happens if `input` does not contain `delimiter`? And what happens with the input part after the last delimiter?

**Comment by tillrohrmann** [comment]
Thanks for your contribution @ajaybhat.

The PR contains many whitespace changes. Would be good to avoid them.

I had a question concerning the implementation. I might be the case that the text after the last delimiter is discarded.

**Comment by fhueske** [comment]
This change breaks the API and should be done with care.
Please add new constructors for the `String` delimiter and deprecate the existing constructors with the `char` parameter. 
Thanks

**Comment by aljoscha** [comment]
@ajaybhat Do you have any updates on this PR?

**Comment by fhueske** [comment]
Hi @ajaybhat, do you plan to update this PR? Otherwise, I'd like to close it.

**Comment by greghogan** [comment]
This PR should be closed as FLINK-2125 was fixed with #2233.

**Pull request closed without merging**